packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -391,23 +392,33 @@ export class DevServerBuilder implements Builder<DevServerBuilderOptions> {
     const architect = this.context.architect;
     const [project, target, configuration] = options.browserTarget.split(':');
 
-    const overrides = {
-      // Override browser build watch setting.
-      watch: options.watch,
-
-      // Update the browser options with the same options we support in serve, if defined.
-      ...(options.optimization !== undefined ? { optimization: options.optimization } : {}),
-      ...(options.aot !== undefined ? { aot: options.aot } : {}),
-      ...(options.sourceMap !== undefined ? { sourceMap: options.sourceMap } : {}),
-      ...(options.vendorSourceMap !== undefined ?
-         { vendorSourceMap: options.vendorSourceMap } : {}),
-      ...(options.evalSourceMap !== undefined ? { evalSourceMap: options.evalSourceMap } : {}),
-      ...(options.vendorChunk !== undefined ? { vendorChunk: options.vendorChunk } : {}),
-      ...(options.commonChunk !== undefined ? { commonChunk: options.commonChunk } : {}),
-      ...(options.baseHref !== undefined ? { baseHref: options.baseHref } : {}),
-      ...(options.progress !== undefined ? { progress: options.progress } : {}),
-      ...(options.poll !== undefined ? { poll: options.poll } : {}),
-    };
+    const overridesOptions: DevServerBuilderOptionsKeys[] = [
+      'watch',
+      'optimization',
+      'aot',
+      'sourceMap',
+      'vendorSourceMap',
+      'evalSourceMap',
+      'vendorChunk',
+      'commonChunk',
+      'baseHref',
+      'progress',
+      'poll',
+    ];
+
+    // remove options that are undefined or not to be overrriden
+    const overrides = Object.keys(options)
+      .reduce<Partial<BrowserBuilderSchema>>(
+        (previous, key: DevServerBuilderOptionsKeys) => (
+          options[key] !== undefined && overridesOptions.includes(key)
```

Comment:
It's simpler to just overload _if_ it's defined, no? The `...(x ? { x } : {})` trick was used because it aligns properly vertically and makes it easier to read, but in this case it makes it harder.

How about this?

```typescript
const overrides = Object.keys(options)
  .filter(key => options[key] !== undefined && overridesOptions.includes(key))
  .reduce((previous, key) => {
    previous[key] = options[key];

    return previous;
  });
```

Reply:
I updated the implementation however, had to change a bit the `reduce`